cpp/include/raft/core/mdspan.hpp

```diff
@@ -312,6 +312,41 @@ RAFT_INLINE_FUNCTION auto unravel_index(Idx idx,
 
 /** @} */
 
+/**
+ * @defgroup mdspan_contiguous Whether the strides imply a contiguous layout.
+ * @{
+ */
+
+/**
+ * @brief Whether the strides imply a c-contiguous layout.
+ */
+template <typename Extents, typename Strides>
+[[nodiscard]] auto is_c_contiguous(Extents const& extents, Strides const& strides) -> bool
+{
+  std::uint64_t stride = 1;
+  for (auto r = extents.rank(); r > 0; r--) {
+    if (stride != strides[r - 1]) { return false; }
```

Comment:
In your test case, in the first iteration isn't `stride = 1` and `strides[r - 1] = 8` where `r = 3`? I don't understand how this doesn't exit in the first iteration itself